packages/docmanager/src/manager.ts

```diff
@@ -572,6 +591,10 @@ namespace Private {
 
   /**
    * A type alias for a standard context.
+   *
+   * #### Notes
+   * We define this as an interface of a specific implementation so that we can
+   * use the implementation-specific functions.
```

Comment:
This code confused me for a while, and finally I saw what was going on and put the note in to help future us understand it. Basically, in the code above, we use the `.factoryName` property that is *not* on the generic IContext interface, but *is* on the Context implementation we use.

One question is if we should just expose the factoryName on the IContext generic interface. Another question is if the manager should just use an attached property to store the factoryName (which I actually did early in this PR before I realized what this code was doing). Of course, another thing to do is to just say that this is a Context, rather than make another alias Private.IContext.

Reply:
Ah, just figured it out. It's because we use the type several places, so it's convenient to have a short name for it. It doesn't have to be an interface, though. It can just be a type alias. I'll change it to a type alias so it's clearer.

Reply:
Well, I guess we *are* using it as an interface, i.e., a type, so it may be fine as-is. For me, it was confusing because I thought it was a generic IContext, and didn't catch it was a new interface from an implementation of IContext. Thoughts?